knack/iframeMapMessengerDev.js

```diff
@@ -0,0 +1,162 @@
+    var script = document.createElement("SCRIPT");
+    script.src =
+      "https://ajax.googleapis.com/ajax/libs/jquery/3.4.1/jquery.min.js";
+    script.type = "text/javascript";
+    document.getElementsByTagName("head")[0].appendChild(script);
+  
+    // Create polling function for made sure jQuery is loaded and ready...
+    var checkReady = function(callback) {
+      if (window.jQuery) {
+        callback(jQuery);
+      } else {
+        window.setTimeout(function() {
+          checkReady(callback);
+        }, 20);
+      }
+    };
+  
+    function AutozoomSendMessageToApp(message) {
+      var iframe = document.getElementById("mapIFrame").contentWindow;
+      const stringifiedMessage = JSON.stringify(message);
+      console.log("autozoom message to app ", message.message)
+      iframe.postMessage(stringifiedMessage, "*");
+    }
+  
+    // Start polling...
+    checkReady(function($) {
+      var $viewSelector = $(myView);
+  
+      // Add React app as iframe if iframe doesn't already exist
+      if ($(myView + " #mapIFrame").length === 0) {
+        https: $(
+          // '<iframe src="https://atd-knack-signs-markings.netlify.app/" frameborder="0" scrolling="yes" id="mapIFrame" \
+          '<iframe src="http://localhost:3000/" frameborder="0" scrolling="yes" id="mapIFrame" \
+      style="width: 100%;height: 523px;"></iframe>'
+        ).appendTo($viewSelector);
+      }
+  
+      function sendMessageToApp(message, iframe) {
+        var stringifiedMessage = JSON.stringify(message);
+        console.log("iFrameMessage to app", message);
+        iframe.postMessage(stringifiedMessage, "*");
+      }
+  
+      // Listen for lat/lon changes
+      // expects a message named "LAT_LON_FIELDS"
+      // uses lat and lng from message to populate input fields in knack
+      window.addEventListener("message", function(event) {
+        if (event.data.source === "react-devtools-content-script") {
+          return;
+        }
+        var data = event.data;
+        if (data.message === "LAT_LON_FIELDS") {
+          var $latLonFields = $("#kn-input-field_3300");
+  
+          $latLonFields.find("#latitude").val(data.lat);
+          $latLonFields.find("[name='longitude']").val(data.lng);
+        }
+      });
+  
+      // Location Details Page Maps
+      function locationDetailsMapMessage(viewId) {
+        var locationViewIFrame = $("#" + viewId + " #mapIFrame")[0].contentWindow;
+        var urlArray = window.location.href.split("/");
+        var recordId = urlArray[urlArray.length - 2];
+        var workOrderId = urlArray[urlArray.length - 4];
+  
+        var markerMessage = {
+          message: "KNACK_LOCATION_DETAILS",
+          view: "view_2733",
+          scene: "scene_1039",
+          token: Knack.getUserToken(),
+          app_id: Knack.application_id,
+          id: recordId,
+          workOrderScene: "scene_1028",
+          workOrderId: workOrderId,
+          workOrderView: "view_2573"
+        };
```

Comment:
I am thinking about moving some of this payload into the react app side, if the view and scenes are hard coded, why not store them in code that is easier to update